scripts/yast2

```diff
@@ -422,6 +422,11 @@ else
 	*) REDO_FILE=/var/lib/YaST2/restart_yast ;;
     esac
     snapshot_pre $module
+
+    # remove the redo file if it already exists before starting the module,
+    # avoid possible infinite loop (bsc#842910)
+    rm -f "$REDO_FILE"
+
```

Comment:
Yes, this should fix only the case when the file already exists *before* starting the module (via `y2base`).

The condition still applies when the module started from `y2base` creates the restart file and wants to restart YaST.